src/material/form-field/form-field.ts

```diff
@@ -473,10 +478,16 @@ export class MatFormField
    * checking every change detection cycle.
    */
   private _initializeOutlineLabelOffsetSubscriptions() {
-    // Whenever the prefix changes, schedule an update of the label offset.
-    this._prefixChildren.changes.subscribe(
-      () => (this._needsOutlineLabelOffsetUpdateOnStable = true),
-    );
+    // Whenever the prefix changes, schedule an update  for label offset.
+    this._prefixChildren.changes.pipe(map(prefixes => !!prefixes.length)).subscribe(hasPrefix => {
```

Comment:
I think we need the `distinctUntilChanged` too, so that we don't get duplicate `true` notifications if the count goes 1 => 2 => 3

Reply:
there could be cases where let's say we have a `form-field` with a `matIconPrefix`, and then we change the prefix to `matTextPrefix`.
In this case we still have only one children (length = 1), but we need to recalculate the label offset.

Reply:
Ah I didn't think about changing between icon and text prefixes, I guess that makes sense.